### ISSUE TITLE
Bio Rifle

### ISSUE THREAD

**Issue by HellDragon-HK** [GreatEmerald/UT3Weapons#9]
- ANIMATION RELATED
  - [ ] The barrel belt that rotates seems to start to rotate and immediately snaps back on UT2004 and looks like it might even be rotating the wrong direction, hard to tell with the snap back it's doing
 - [ ] 3rd person primary fire animation loop missing

 - GAME PLAY RELATED
  - [ ] Alt fire is instant hit rather than the cling to target slow damage, guessing this isn't possible with UT2004 though
 - [ ] Weapon pickup is really close to the weapon base pad, on UT3 it floats quite a bit higher
 - [ ] The projectile on UT3 feels larger and looks larger when on a surface
 - [ ] A single primary fire shot projectile stays on a surface longer in UT3 than in UT2004 by 2 seconds or more
- [ ] Alt fire projectile splits into pieces like it should but always remains small, on UT3 it always remains large
 - [ ] Alt fire projectile lingers longer on UT3
 - [ ] In 3rd person the alt fire glob is close enough to correct size, weird it seems smaller in 1st person, however upon hitting a surface it becomes the same size as a primary fire shot which it doesn't in UT3
 - [ ] Berserk in UT2004 takes about 3 seconds to fire 10 primary fire shots and in UT3 it takes not quite 3 seconds it feels like, UT3 feels faster still but not by much

 - SOUND RELATED
  - [ ] Primary fire sounds completely different in UT2004, I am taking into consideration it may be the difference between my TV and PC speakers however and need another opinion
  - [ ] Alt fire impact sound is almost like Unreal 1's GES Bio Rifle in UT2004 it sounds like, that isn't right by any means and sounds like multiple explosions instead of a sticky gooey slime

 - VISUAL EFFECTS RELATED
 - [ ] Muzzle flash on UT3 is quite a bit different and even includes small globs of biowaste
 - [ ] UT2004 projectile is well just odd, for one it's really bright and black in the center, on UT3 the whole thing is a dull green and slightly black in the center and looks more like slime....I would swear this is the old Unreal 1 GES Bio Rifle projectile
 - [ ] The projectile when on a surface in UT3 has green smoke coming from it and also appears to be bubbling.....I just briefly saw thin white smoke in UT2004 but the projectile explodes so much quicker than UT3 there isn't enough time to study it
 - [ ] Alt fire charge up is missing the giant biowaste ball in the center of the weapon
 - [ ] Alt fire leaves a large visual impact for a bit on UT3 after it explodes and is no longer a threat

**Comment by GreatEmerald** [comment]
In theory you could make the projectile cling and siphon health, but it would need a lot of new code.

The blob appearance is most likely a 64-bit bug. It's quite true to the original otherwise, I spent a lot of time on it.

The snap back of the animation is an artefact of how the UT2004 animation system works. I'm not sure whether this can be changed (aside from making it instant and thus jerky, or making it slower to go back, which would also be jerky if you try to fire again right after).

The alt fire blob on screen is literally the projectile in UT3. There's no easy way of making it appear in UT2004 I don't think.

**Comment by HellDragon-HK** [comment]
So then the projectile has a drawscale problem too?

**Comment by GreatEmerald** [comment]
No, I meant that in UT3 it's not part of the weapon, it's spawned on the fly. If I tried to do that in UT2004 it would not become bigger over time and most likely deal damage to yourself.

**Comment by HellDragon-HK** [comment]
So then not much else can be done then and it can be wrote off already?

**Comment by GreatEmerald** [comment]
With a lot of new code it could be done, but it's not worth it.

**Comment by HellDragon-HK** [comment]
Maybe I've got a mod that already has something similar, I'll keep an open mind on it and let you know if I find anything

**Comment by HellDragon-HK** [comment]
More inconsistent damage and timing as well here
Fired at the ground in UT3 a small primary fire shot it exploded as soon as it hit the ground and in UT2004 it stuck to it and took time to explode even though I was standing directly on it
UT3 - 100 to 90 and in UT2004 100 - 84  this was with a single tap of primary fire

**Comment by HellDragon-HK** [comment]
UT3 Manta and Scorpion took 26 damage from single primary, in UT2004 they took 31-33 and Manta gets extreme knockback

In UT3 alt did 100 to both Manta and Scorpion consistently, in UT2004 Manta was destroyed on both attempts and Scorpion took 46-57, charged alt does not push vehicles in UT3 but does in UT2004

**Comment by HellDragon-HK** [comment]
I learned that the sounds for the alt fire are right in UT2004 but only the main glob uses the explode sound in UT3 while all the smaller spawn from it use the normal primary explode sound, currently in UT2004 they are all using the alt fire explode sound.

There are also 2 other sound issues, the alt fire has a second not fully charged sound that isn't in use anywhere and the alt fire charged loop sound does not work.

**Comment by HellDragon-HK** [comment]
Bio Rifle 3P animation mesh is completely missing, I can see it in the UT3 editor but nowhere in the weapons ukx file.   How does that work in-game if only the 1P mesh is there anyway?

As for the suspended glob, I don't think it's actually the projectile we're seeing, instead it's just a visual effect and the glob is spawned the same way as it's always been I think, it looks quite a bit different from the projectile itself at any rate.

Are you sure the projectiles are the UT3 mesh? There is a single mesh with 2 animations in the UT3 editor but I'm not finding it in the ukx anywhere.

**Comment by HellDragon-HK** [comment]
Alt Fire charged ambient loop does actually work but is insanely quiet, with music completely off I barely noticed it was there....any idea on why it would be doing that?
